src/lib/arch/riscv/ExceptionHandler.cc

```diff
@@ -633,14 +634,71 @@ bool ExceptionHandler::init() {
         stateChange = {ChangeType::REPLACEMENT, {R0}, {0ull}};
         break;
       }
-
       default:
         printException(instruction_);
         std::cout << "\n[SimEng:ExceptionHandler] Unrecognised syscall: "
                   << syscallId << std::endl;
         return fatal();
     }
 
+    return concludeSyscall(stateChange);
+  } else if (exception == InstructionException::AtomicOperation) {
+    // TODO this may complete in only a single cycle and not the value from the
+    // config file. The instruction should be completely reissued and allowed to
+    // execute with latency
```

Comment:
Yes, my thought with the comment is that it will apply the latency when going through the EU. This is when the instruction is logically only raising the exception and while other instructions are being executed OoO at the same time. The instruction should be performing its actual logic once the pipeline has been "serialised" (once it is at the head of the ROB). In the code this happens in the exception handler which doesn't (as far as I am aware) account for the latency. It should only be a minor accuracy inconsistency, but an inconsistency none the less

Reply:
I see. Maybe the comment prefix should be `NOTE` then rather than `TODO`?